### PR TITLE
Bump kube-vip image v0.6.0 and deprecate kube-vip cloud provider

### PR DESCRIPTION
**Pull request by yaocw2020**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

**Solution:**

- Bump kube-vip image v0.6.0 for to enable service security
- Remove kube-vip cloud provider because it'll be replaced by IPPool

**Related Issue:** https://github.com/harvester/harvester/issues/3132

**Test plan:**

### CODE REVIEW

File: pkg/controller/master/rancher/register.go [hunk 1]
```diff
@@ -143,22 +152,35 @@ func (h *Handler) registerExposeService() error {
...
+				svc.Annotations[keyKubevipLoadBalancerIPs] = vip.IP
 			}
 		}
 
 		if _, err := h.Services.Create(svc); err != nil {
 			return err
 		}
 	}
+	// Update annotations of the ingress-expose service created before Harvester v1.2.0
+	if svc.Annotations == nil || svc.Annotations != nil && svc.Annotations[keyKubevipIgnoreServiceSecurity] != trueStr {
```
Comment (guangbochen) [inline pkg/controller/master/rancher/register.go]: you should be able to check it directly via 
`if svc.Annotations == nil || svc.Annotations[keyKubevipIgnoreServiceSecurity] != trueStr `,
also, please help to address the CodeFactor warning.

**Review by guangbochen — APPROVED** [review]
LGTM, and please help to address the PR conflict; thanks.

File: deploy/charts/harvester/values.yaml [hunk 2]
```diff
@@ -457,16 +457,21 @@ harvester-load-balancer:
 kube-vip:
   enabled: false
   image:
-    repository: ghcr.io/kube-vip/kube-vip
-    tag: v0.5.12
+    repository: ghcr.io/kube-vip/kube-vip-iptables
+    tag: v0.6.0
```
Comment (chrisho) [inline deploy/charts/harvester/values.yaml:462]: What are the differences between `ghcr.io/kube-vip/kube-vip` and `ghcr.io/kube-vip/kube-vip-iptables`? 
I have checked the images the `version` display Build and Version info are the same.
```

❯ docker run --rm ghcr.io/kube-vip/kube-vip-iptables:v0.6.0 version
Kube-VIP Release Information
Version:  v0.6.0
Build:    b594d459fc4e2061015157715fc84e84bee12d81
❯ docker run --rm ghcr.io/kube-vip/kube-vip:v0.6.0 version
Kube-VIP Release Information
Version:  v0.6.0
Build:    b594d459fc4e2061015157715fc84e84bee12d81
```

Comment (yaocw2020) [reply deploy/charts/harvester/values.yaml:462]: The image ghcr.io/kube-vip/kube-vip-iptables is built from https://github.com/kube-vip/kube-vip/blob/main/Dockerfile_iptables, while ghcr.io/kube-vip/kube-vip from https://github.com/kube-vip/kube-vip/blob/main/Dockerfile. 

 ghcr.io/kube-vip/kube-vip-iptables: sha256:aab8a321a08726d2b9b49eedfdb23168f11adb55f0c21dcfbda3b48bc9af74bd
 ghcr.io/kube-vip/kube-vip: sha256:defef5b817487ca1f30123df5cfd17bad996d0b5f885e3004541810986d132d2

**Review by chrisho — APPROVED** [review]
LGTM, thanks.

**Review by futuretea — APPROVED** [review]
LGTM. Thanks

**Pull request merged**